### DISCUSSION TITLE
Is there any api to get all issues associated with a given list of commit

### DISCUSSION THREAD

**Discussion by mdmdakbar** [community/community#24541] [Apps, API and Webhooks]
<p>Given a list of commits, I want to get all the issues linked to those commits.</p>
<p>Is there any rest or graphql api available today to fetch above information?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There isn’t an API for this because it is done by <a href="https://help.github.com/articles/autolinked-references-and-urls/#issues-and-pull-requests" rel="nofollow noopener">pattern recognition in the commit comment</a>. When a commit is pushed that references an issue or PR, <a href="https://developer.github.com/v3/issues/events/" rel="nofollow noopener">an event is added to the issue</a> to refer back to the commit but no such metadata is created for the commit.</p>

**Comment by mdmdakbar** [comment]
<p><a href="https://help.github.com/articles/searching-issues-and-pull-requests/#search-by-commit-sha" rel="nofollow noopener">https://help.github.com/articles/searching-issues-and-pull-requests/#search-by-commit-sha</a></p>
<p>The above api says "<em>If you know the specific SHA hash of a commit, you can use it to search for pull requests that contain that SHA."</em></p>
<p>Isn’t there any way to get issues using search api? I think pull request are treated as issues only by GitHub.</p>

**Comment by lee-dohm** [comment]
<p>There are ways to get a list of issues via the search API but not via a commit SHA, no, for the reasons I stated before.</p>

**Comment by alexortizrosado** [comment]
<p><em>(Posting this here in case someone else finds it useful as I was having a hard time figuring out how to get Issues linked to a Pull Request)</em></p>
<p>There’s no direct way of getting a list of <strong>Issues</strong> linked to a <strong>Pull Request</strong>.</p>
<p>But each <strong>Pull Request</strong> includes a <em>timeline of events</em> and one of those events is when an <strong>Issue</strong> gets linked. Using the <em>timeline of events</em> I was able to write a GitHub APIv4 API request and some javascript to get the Issues linked to a PR:</p>
<p>First, here’s the GraphQL query:</p>
<pre><code>{
  resource(url: "https://github.com/Org/Project/pull/1234") {
    ... on PullRequest {
      timelineItems(itemTypes: [CONNECTED_EVENT, DISCONNECTED_EVENT], first: 100) {
        nodes {
          ... on ConnectedEvent {
            id
            subject {
              ... on Issue {
                number
              }
            }
          }
          ... on DisconnectedEvent {
            id
            subject {
              ... on Issue {
                number
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>If you run this in the GitHub GraphQL Explorer (<a href="https://developer.github.com/v4/explorer/" rel="nofollow noopener">https://developer.github.com/v4/explorer/</a>), you’ll see all the events that an Issue was connected and disconnected to a Pull Request.</p>
<p>Using that query, I pass the response to this code that I wrote for a nodejs app that then determines which <strong>Issue</strong> is <em>still</em> linked to the <strong>Pull Request</strong></p>
<pre><code>const issues = {};
resource.timelineItems.nodes.map(node =&gt; {
    if (issues.hasOwnProperty(node.subject.number)) {
        issues[node.subject.number]++;
    } else {
        issues[node.subject.number] = 1;
    }
});
const linkedIssues = [];
for (const [issue, count] of Object.entries(issues)) {
    if (count % 2 != 0) {
        linkedIssues.push(issue);
    }
}
console.log(issues);

console.log(linkedIssues);
</code></pre>
<p>The logic here is as follows:</p>
<ol>
<li>Get a list of all events on a Pull Request of the type CONNECTED_EVENT and DISCONNECTED_EVENT</li>
<li>Create a map, keyed by Issue number and keep a count of how may times the issue is CONNECTED and DISCONNECTED</li>
<li>From that map, look for keys that have an odd-numbered count, as these are the events that have been CONNECTED that don’t have a corresponding DISCONNECTED event.</li>
</ol>
<p>It’s not a <em>super elegant</em> solution, but it solves what I need, which was to find those Linked issues.</p>
<p>Hopefully this helps someone else out </p>

**Comment by anrico-oltmanns** [comment]
In our organization, we merge everything to "develop". Once a week or bi-weekly we merge everything from "develop" to "master". We want to have a list of all the issues associated with the PR in the PR description based on the commits, which are linked to the issues (with #issue-number).

Is it possible? Ive tried a lot of things but nothing seemed to work. But im not to experienced with the Github Rest API
